### PR TITLE
cli: `vote-authorize-voter` no longer fails if the current authorized voter is not the fee payer

### PR DESCRIPTION
**Pull request by mvines**
`solana vote-authorize-voter` was assuming that the current authorized voter is the fee payer, but if the current authorized voter is the vote account itself then it cannot pay the fee and it's not possible to change the authorized voter (same for authorized withdrawer)

### CODE REVIEW

File: cli/src/vote.rs [hunk 1]
```diff
@@ -80,14 +80,22 @@ impl VoteSubCommands for App<'_, '_> {
                         .index(1)
                         .value_name("VOTE_ACCOUNT_ADDRESS")
                         .required(true),
-                        "Vote account in which to set the authorized voter. "),
+                        "Vote account in which to set the authorized voter."),
```
Comment (CriesofCarrots) [inline cli/src/vote.rs:83]: Fyi, I think the new `pubkey!` macro expects these spaces.

Comment (CriesofCarrots) [reply cli/src/vote.rs:83]: Or I should say, this text will run together without them

Comment (mvines) [reply cli/src/vote.rs:83]: Oh weird, we couldn't hide that in the macro?

I'll restore them here though

Comment (mvines) [inline cli/src/vote.rs:83]:
```suggestion
                        "Vote account in which to set the authorized voter. "),
```

File: cli/src/vote.rs [hunk 2]
```diff
@@ -80,14 +80,22 @@ impl VoteSubCommands for App<'_, '_> {
...
+                        .validator(is_valid_signer)
+                        .help("Current authorized vote signer."),
+                )
+                .arg(
+                    pubkey!(Arg::with_name("new_authorized_pubkey")
+                        .index(3)
+                        .value_name("NEW_AUTHORIZED_PUBKEY")
                         .required(true),
-                        "New authorized vote signer. "),
+                        "New authorized vote signer."),
```
Comment (mvines) [inline cli/src/vote.rs:98]:
```suggestion
                        "New authorized vote signer. "),
```

File: cli/src/vote.rs [hunk 3]
```diff
@@ -98,14 +106,22 @@ impl VoteSubCommands for App<'_, '_> {
                         .index(1)
                         .value_name("VOTE_ACCOUNT_ADDRESS")
                         .required(true),
-                        "Vote account in which to set the authorized withdrawer. "),
+                        "Vote account in which to set the authorized withdrawer."),
```
Comment (mvines) [inline cli/src/vote.rs:109]:
```suggestion
                        "Vote account in which to set the authorized withdrawer. "),
```

File: cli/src/vote.rs [hunk 4]
```diff
@@ -98,14 +106,22 @@ impl VoteSubCommands for App<'_, '_> {
...
+                        .validator(is_valid_signer)
+                        .help("Current authorized withdrawer."),
+                )
+                .arg(
+                    pubkey!(Arg::with_name("new_authorized_pubkey")
+                        .index(3)
                         .value_name("AUTHORIZED_PUBKEY")
                         .required(true),
-                        "New authorized withdrawer. "),
+                        "New authorized withdrawer."),
```
Comment (mvines) [inline cli/src/vote.rs:124]:
```suggestion
                        "New authorized withdrawer.v"),
```

**Comment by mvines** [comment]
gui coding ftw

File: cli/src/vote.rs (hunk 1 above)
Comment (CriesofCarrots) [reply cli/src/vote.rs]: I assume we could

**Pull request merged**